networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/encoding/SnappyBlockCompressor.java

```diff
@@ -24,16 +24,24 @@
  */
 public class SnappyBlockCompressor {
 
-  public Bytes uncompress(final Bytes compressedData, final SszLengthBounds lengthBounds)
+  public Bytes uncompress(
```

Comment:
We should also add a check to `compress` to ensure that the uncompressed size is <= `maxBytesLength`.

> Likewise, clients MUST NOT emit or propagate messages larger than this limit.

https://github.com/ethereum/consensus-specs/blob/3d180786af8f661a9dc53d1aadf0f89972a56024/specs/phase0/p2p-interface.md?plain=1#L276